src/tags-input.js

```diff
@@ -53,12 +59,17 @@ tagsInput.directive('tagsInput', function($timeout, $document, tagsInputConfig,
 
         tagIsValid = function(tag) {
             var tagText = getTagText(tag);
+            var valid = tagText &&
+                     tagText.length >= options.minLength &&
+                     tagText.length <= options.maxLength &&
+                     options.allowedTagsPattern.test(tagText) &&
+                     !tiUtil.findInObjectArray(self.items, tag, options.displayProperty);
+
+            if (scope.onTagAdding()) {
+                valid = valid && scope.onTagAdding()(tag);
```

Comment:
The `onTagAdding` callback should be called this way: `onTagAdding({ $tag: tag }`, so a variable named `$tag` is available to be used in the expression provided by the user.